Tools/ardupilotwaf/chibios.py

```diff
@@ -18,6 +18,10 @@
 import base64
 import subprocess
 
+# modify our search path:
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), '../../libraries/AP_HAL_ChibiOS/hwdef/scripts'))
+import chibios_hwdef
```

Comment:
Since this was written we sprouted hwdefs for the Linux HAL.  It does what you're doing in here, and it does work well!